contrib/lang/agda/extensions.el

```diff
@@ -0,0 +1,48 @@
+;;; extensions.el --- Agda2 Layer extensions File for Spacemacs
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2015 Oliver Charles & Contributors
+;;
+;; Author: Oliver Charles <ollie@ocharles.org.uk>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defvar agda-pre-extensions '(agda))
+
+(defun agda/init-agda ()
+  (custom-set-faces
+   '(agda2-highlight-datatype-face ((t (:inherit font-lock-type-face))))
+   '(agda2-highlight-function-face ((t (:inherit font-lock-type-face))))
+   '(agda2-highlight-inductive-constructor-face ((t (:inherit font-lock-function-name-face))))
+   '(agda2-highlight-keyword-face ((t (:inherit font-lock-keyword-face))))
+   '(agda2-highlight-module-face ((t (:inherit font-lock-constant-face))))
+   '(agda2-highlight-number-face ((t nil)))
+   '(agda2-highlight-postulate-face ((t (:inherit font-lock-type-face))))
+   '(agda2-highlight-primitive-type-face ((t (:inherit font-lock-type-face))))
+   '(agda2-highlight-record-face ((t (:inherit font-lock-type-face)))))
+
+  (load-file (let ((coding-system-for-read 'utf-8))
+               (shell-command-to-string "agda-mode locate")))
```

Comment:
It's weird, but yeah, that is apparently the recommended way of loading agda-mode, since it's not on elpa.